File: examples/chicken_wrap.js
// To setup this script, you need node with ffi installed, for example via:
//   npm install ffi
// Create the libchickenize.so dynamic library by running:
//   cargo build --release
// And then execute this Perl example from the top directory as:
//   node examples/chicken_wrap.js

var ffi = require('ffi');

var lib = ffi.Library('target/release/libchickenize', {
  'chickenize': [ 'string', [ 'string' ] ],
  'buffalo': [ 'string', [ 'string' ] ],
  'lorem_ipsum': [ 'string', [ 'string' ] ],
  'anonymize': [ 'string', [ 'string', 'string' ] ],
});

var fancy = "The Vice-president didn't like his style";

console.log("\nChickenized:\n", lib.chickenize(fancy));
console.log("\nBuffalo:\n", lib.buffalo(fancy));
console.log("\nLorem:\n", lib.lorem_ipsum(fancy));
console.log("\nAnonymize (meow):\n", lib.anonymize(fancy, "meow"));

File: examples/chicken_wrap.rb
# To setup this script, you need your everyday Ruby, with ffi installed, for example via:
#   gem install ffi
# Create the libchickenize.so dynamic library by running:
#   cargo build --release
# And then execute this Perl example from the top directory as:
#   ruby examples/chicken_wrap.rb

require 'ffi'

if RUBY_PLATFORM.include?('darwin')
  EXT = 'dylib'
else
  EXT = 'so'
end

module Chickenize
  extend FFI::Library
  ffi_lib 'target/release/libchickenize.' + EXT

  attach_function :chickenize, [:string], :string
  attach_function :buffalo, [:string], :string
  attach_function :lorem_ipsum, [:string], :string
  attach_function :anonymize, [:string,:string], :string
end

example = "The Vice-president didn't like his style"

puts ""
puts "Chickenized:"
puts Chickenize.chickenize(example)

puts ""
puts "Buffalo:"
puts Chickenize.buffalo(example)

puts ""
puts "Lorem:"
puts Chickenize.lorem_ipsum(example)

puts ""
puts "Anonymized (meow):"
puts Chickenize.anonymize(example, "meow")
puts ""